components/domains/externalDomainTransferModal.tsx

```diff
@@ -0,0 +1,168 @@
+import { InputAdornment, Modal, TextField } from "@mui/material";
+import { useContractWrite } from "@starknet-react/core";
+import { useRouter } from "next/router";
+import React, {
+  FunctionComponent,
+  useState,
+  useEffect,
+  useContext,
+} from "react";
+import styles from "../../styles/components/modalMessage.module.css";
+import { isHexString, minifyAddress } from "../../utils/stringService";
+import Button from "../UI/button";
+import { utils } from "starknetid.js";
+import { StarknetIdJsContext } from "../../context/StarknetIdJsProvider";
+import ConfirmationTx from "../UI/confirmationTx";
+import { useNotificationManager } from "../../hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "../../utils/constants";
+import { Call } from "starknet";
+
+type ExternalDomainsTransferModalProps = {
+  domain: string;
+  domainEncoded: string;
+  resolverContract: string;
+  handleClose: () => void;
+  isModalOpen: boolean;
+  domainKind?: string;
+};
+
+const ExternalDomainsTransferModal: FunctionComponent<
+  ExternalDomainsTransferModalProps
+> = ({
+  domainEncoded,
+  resolverContract,
+  handleClose,
+  isModalOpen,
+  domainKind,
+}) => {
+  const router = useRouter();
+  const { externalDomain: domain } = router.query;
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const { addTransaction } = useNotificationManager();
+  const [addressInput, setAddressInput] = useState<string>("");
+  const { starknetIdNavigator } = useContext(StarknetIdJsContext);
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [callData, setCallData] = useState<Call[]>([]);
+
+  const { writeAsync: transfer_name, data: transferData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    setCallData([
+      {
+        contractAddress: resolverContract,
+        entrypoint: "transfer_name",
+        calldata: [domainEncoded, targetAddress],
+      },
```

Comment:
Could we add this call data and the set as main domain for external domains to special file as we have for registerCalls.ts for example ?

Reply:
did you forget this comment ?

Reply:
Ah yes, I forgot about this.